src/main/java/com/psk/pyragas/ToolRent/usecases/AddItem.java

```diff
@@ -18,34 +16,36 @@
 @Model
 public class AddItem implements Serializable {
 
-    @Inject
-    private ItemsDAO itemsDAO;
-
-    @Inject
-    private AdvertisementsDAO advertisementsDAO;
-
-    @Getter @Setter
-    private Item itemToCreate = new Item();
-
-    @Getter @Setter
-    private Advertisement advertisement;
-
-    @PostConstruct
-    public void init() {
-        Map<String, String> requestParameters =
-                FacesContext.getCurrentInstance().getExternalContext().getRequestParameterMap();
-
-        this.advertisement = advertisementsDAO.findOne(Long.parseLong(requestParameters.get("advertisementId")));
-    }
-
-    @Transactional
-    public String createItem() {
-
-        itemToCreate.setAdvertisement(this.advertisement);
-        itemsDAO.persist(itemToCreate);
-
-        return "add_item.xhtml?faces-redirect=true&advertisementId=" + this.advertisement.getId();
-    }
+//    TODO: fix this use case
+//
+//    @Inject
+//    private ItemsDAO itemsDAO;
+//
+//    @Inject
+//    private AdvertisementsDAO advertisementsDAO;
+//
+//    @Getter @Setter
+//    private Item itemToCreate = new Item();
+//
+//    @Getter @Setter
+//    private Advertisement advertisement;
+//
+//    @PostConstruct
+//    public void init() {
+//        Map<String, String> requestParameters =
+//                FacesContext.getCurrentInstance().getExternalContext().getRequestParameterMap();
+//
+//        this.advertisement = advertisementsDAO.findOne(Long.parseLong(requestParameters.get("advertisementId")));
+//    }
+//
+//    @Transactional
+//    public String createItem() {
+//
+//        itemToCreate.setAdvertisement(this.advertisement);
+//        itemsDAO.persist(itemToCreate);
+//
+//        return "add_item.xhtml?faces-redirect=true&advertisementId=" + this.advertisement.getId();
+//    }
```

Comment:
I guess it's better to leave this code. We might need it for reference when implementing Decorators.

Reply:
Yes, I agree. I left it commented out for this reason.